File: solutions/haskell/space-age/1/src/SpaceAge.hs
module SpaceAge (Planet(..), ageOn) where

data Planet = Mercury
            | Venus
            | Earth
            | Mars
            | Jupiter
            | Saturn
            | Uranus
            | Neptune

ageOn :: Planet -> Float -> Float
ageOn planet seconds = case planet of
  Earth -> (seconds / 31557600)
  Mercury -> ageOn Earth (seconds / 0.2408467)
  Venus -> ageOn Earth (seconds / 0.61519726)
  Mars -> ageOn Earth (seconds / 1.8808158)
  Jupiter -> ageOn Earth (seconds / 11.862615)
  Saturn -> ageOn Earth (seconds / 29.447498)
  Uranus -> ageOn Earth (seconds / 84.016846)
  Neptune -> ageOn Earth (seconds / 164.79132)

File: solutions/typescript/space-age/1/space-age.ts
export function age(planet: string, seconds: number): number {
  return Math.round(100 * (seconds / 31557600) / (map.get(planet) ?? 1)) / 100
}

const map = new Map([
  ['mercury', 0.2408467],
  ['venus', 0.61519726],
  ['earth', 1.0],
  ['mars', 1.8808158],
  ['jupiter', 11.862615],
  ['saturn', 29.447498],
  ['uranus', 84.016846],
  ['neptune', 164.79132]
])
